src/python/zquantum/core/estimator.py

```diff
@@ -129,17 +152,46 @@ def get_estimated_expectation_values(
             frame_circuits.append(circuit + frame_circuit)
             frame_operators.append(frame_operator)
 
-        if n_samples is not None:
-            logger.warning(
-                f"""Using n_samples={n_samples} (argument passed to get_estimated_expectation_values). 
-                    Ignoring backend.n_samples={backend.n_samples}"""
+        if shot_allocation_strategy == "uniform":
+
+            if n_total_samples is not None:
+                raise ValueError(
+                    "Uniform sampling does not yet support n_total_samples."
+                )
+
+            if n_samples is not None:
+                logger.warning(
+                    f"""Using n_samples={n_samples} (argument passed to get_estimated_expectation_values). 
+                        Ignoring backend.n_samples={backend.n_samples}"""
+                )
+                n_samples = (n_samples,) * len(frame_circuits)
+                measurements_set = backend.run_circuitset_and_measure(
+                    frame_circuits, n_samples
+                )
+            else:
+                measurements_set = backend.run_circuitset_and_measure(frame_circuits)
+
+        elif shot_allocation_strategy == "optimal":
+            if n_total_samples is None:
+                raise ValueError(
+                    "For optimal shot allocation, n_total_samples must be provided."
+                )
+
+            if n_samples is not None:
+                raise ValueError(
+                    "Optimal shot allocation does not support n_samples; use n_total_samples instead."
+                )
+
+            K2, nterms, measurements_per_frame = estimate_nmeas_for_frames(
```

Comment:
What's the `K2` ? It doesn't seem to be used anywhere?

Reply:
It's the proportionality constant between number of measurements and inverse square precision; see [docstring](https://github.com/zapatacomputing/z-quantum-core/blob/f9a3de0c9a7cd4c654d928a03ad52062cebf3dce/src/python/zquantum/core/hamiltonian.py#L275). Anyway I replaced with `_` since it's not used here.